pkg/controllers/inventory/inventory_controller.go

```diff
@@ -342,15 +343,23 @@ func (r *ReconcileBareMetalAsset) checkAssetSecret(ctx context.Context, instance
 	})
 
 	// Set BaremetalAsset instance as the owner and controller
+	modified := false
 	if secret.OwnerReferences == nil || len(secret.OwnerReferences) == 0 {
+		modified = true
 		if err := controllerutil.SetControllerReference(instance, secret, r.scheme); err != nil {
 			klog.Errorf("Failed to set ControllerReference, %v", err)
 			return err
 		}
-		if err := r.client.Update(ctx, secret); err != nil {
-			klog.Errorf("Failed to update secret with OwnerReferences, %v", err)
-			return err
-		}
+	}
+
+	secretTypeLabel := map[string]string{
+		"cluster.open-cluster-management.io/backup": "",
```

Comment:
looks good; it's optional but if you want to tag the label to know it's used by a BMA asset, then you can add the value "bma" as you did initially with the other label

cluster.open-cluster-management.io/backup: "bma"